rfc/rfc000-package-server.md

```diff
@@ -1,19 +1,87 @@
+# Package Server
+
 ---
+
 RFC: 000
-Author: Peter Bourgon <peter@bourgon.org>
+
+Author: Matt Heath <matt@mattheath.com>
+
 Status: Draft
----
 
-# package server
+---
 
 ## Motivation
 
-http://peter.bourgon.org/go-kit/#package-server
+[http://peter.bourgon.org/go-kit/#package-server](http://peter.bourgon.org/go-kit/#package-server)
+
+Package server is probably the biggest and most important component of the toolkit. Ideally, we should be able to write our services as implementations of normal, nominal Go interfaces, and delegate integration with the environment to the server package. The package should encode and enforce conventions for server-side concerns, like health checks, system-wide request tracing, connection management, backpressure and throttling, and so on. For each of those topics, it should provide interfaces for different, pluggable strategies. It should integrate with service discovery, and work equally well over multiple transports. Considerable prior art exists in the form of Finagle, Karyon (Netflixs application service library), and likely many more.
 
 ## Scope
 
-To be defined.
+### Endpoints
+
+*   An endpoint is defined as a handler interface which receives a request and returns a response or an error to the client.
+*   A server SHALL have one or more endpoints.
```

Comment:
Somewhat, yes... it just gets confusing because, e.g., the health for one endpoint may have nothing to do with the health of another (I can give a specific example if this doesn't make sense), so there's something to be said for allowing multiple Servers per process in order to avoid these sorts of shared-fate issues.